server/src/main/java/com/flipkart/varadhi/auth/DefaultAuthorizationProvider.java

```diff
@@ -0,0 +1,139 @@
+package com.flipkart.varadhi.auth;
+
+import com.flipkart.varadhi.entities.UserContext;
+import com.flipkart.varadhi.utils.YamlLoader;
+import io.vertx.core.Future;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.Pair;
+
+
+@Slf4j
+public class DefaultAuthorizationProvider implements AuthorizationProvider {
+    private DefaultAuthorizationConfiguration configuration;
+    private volatile boolean initialised = false;
+
+    @Override
+    public Future<Boolean> init(AuthorizationOptions authorizationOptions) {
```

Comment:
Was planning to read the file as json object and pass to the providers. But, decided against it because I think it would be better to have control at the provider end in case they wish to specify the configuration as JSON instead of YAML or even interpret the configFile path as an URI and fetch the config over the internet or file server.

Reply:
with this we can also give configFile as:
`configFile: "https://raw.githubusercontent.com/acme-org/acme-project/customconfig.json"`

And the provider can then make an http call to fetch and parse the config.